### ISSUE TITLE
New pattern - Opt for accessible version

### ISSUE THREAD

**Issue by jgiardino** [patternfly/patternfly#1882]
For components that are not keyboard and screen reader accessible, we need a pattern that allows the user to opt out of the inaccessible version and use the accessible version of the UI instead.

The only case for this that we have now is the [table with virtual scroll](https://2011-pr-patternfly-react-patternfly.surge.sh/patternfly-4/virtual%20scroll/virtualized/), which is inaccessible currently (and even if it could be made accessible, it would be a lower quality user experience). The version of this component that is more accessible to screen reader and keyboard-only users is a paginated table. 

This pattern would be similar to this pattern would be a [SkipToContent](https://pf4.patternfly.org/components/SkipToContent/examples/) component, in that the action is visible on focus, or available to screen reader users, but not ever visible when just using a mouse. 

Additionally, should this type of pattern include a message area for an additional explanation?

For example, if the button that becomes visible says "Enable Pagination" would the user otherwise know that they would not be able to interact with the table without enabling pagination? Or should we include a message with the button that says "Pagination provides an alternate method of accessing contents in this view which might be easier to use when navigating with a keyboard" (ugh, wording)?

**Comment by jgiardino** [comment]
Here is a [quick (i.e. rough) codepen](https://codepen.io/jenngiardino/pen/wbZwKx) to demonstrate the interaction that I was thinking. If using chrome, click "Colors" in the nav sidebar to set focus on the page, then hit Tab. The next item to receive focus is a button labeled "Use Alternate".

It would be good to get design input on what the button label should be, and if we should include an additional messaging so that the user understands what's going on. And also, do the styles we use for the "Skip to content" link work for this pattern, or do we need additional styling?

**Comment by priley86** [comment]
+1 - thanks for starting this convo @jgiardino !

I have noted some gaps as far as being able to "page" in a virtual scrolling table. Currently I do not see a good way of doing this because w/o modifications, you would not have any keyboard interaction to emulate this paging behavior the same way as you would with a paginated table.

That being said, I think it from an OpenShift perspective I see two major variations on this topic:

1. Resource tables - these are expected to be long scrolling tables and the current UX is virtual scroll for existing performance reasons. This data also changes frequently (connected via web socket) so it was determined that pagination w/ alerts was not a viable option either (worth considering). I think on these tables, it would be nice to consider the workaround above mentioned (for toggling to a paginated variation that is accessible) or considering an "infinite scrolling" pattern where performance is not deemed as critical. Through this effort and research, we found that virtual scrolling adds a pretty significant performance boost when rendering and scrolling more than 1k rows, however that is not to say that one could not just append rows as the user scrolls downward (rather than virtually hiding them as the user scrolls) if the list is never anticipated to go above 10k or so this may be perfectly fine. My point is, one could easily also add buttons or paging controls for infinite scrolling table and I believe that `aria` and vanilla PF4 table markup would be accessible and work just fine if you had some way of announcing the total row count. This is not the case w/ virtual scroll where rows are hidden from view. I believe the closest accessible example of this I can find is in the following w3 spec (Example 3: Scrolling Data Grid w/ Column Hiding):
https://www.w3.org/TR/wai-aria-practices-1.1/examples/grid/dataGrids.html

So in summary, I am just proposing that we actually classify these existing resource tables and see if we can't consider pagination or infinite scroll for some of them for sighted users and unsighted users. This may lead to less code maintenance. It would also be interesting to consider a "PF UX Decision Tree" for Tables if we wanted...

2. Detail table views - these views are expected to have smaller data sets and vanilla PF4 markup applies (no paging, infinite scroll, or virtual scrolling generally applies). These should behave like our other PF4 tables.

This is my current thoughts on this topic after some initial discussions. Would be glad to hear more feedback from @janwright73, @beanh66, @matthewcarleton, and @andybraren though!

**Comment by priley86** [comment]
Some additional notes here (and just recalling from previous discussions) that may prove relevant to the solution, some product APIs may never move towards pagination or have plans to move their backend APIs to support a paginated format (such as kubernetes APIs). This would not remove the possibility of doing client side pagination on all data present to the user, but that could lead to other performance issues if the data set is quite large and we aren't doing server side paging w/ the API. Thus, it may still prove useful to consider an infinite scrolling or "Scrolling Data Grid w/ Column Hiding" type solution noted above.

cc: @rhamilto

**Comment by priley86** [comment]
linking OpenShift [JIRA issue 1599](https://jira.coreos.com/browse/CONSOLE-1599)

Probably makes sense to reach a conclusion here first. cc: @jgiardino

**Comment by mcoker** [comment]
cc @jessiehuff
